File: hive-financial-control/src/server/services/integrations/pagacuotas-links.ts
export function normalizePagaCuotasPortalLink(link: string | null | undefined) {
  if (!link) return null;

  try {
    const url = new URL(link);

    if (url.pathname === "/client/payment") {
      url.pathname = "/client/portal";
      url.search = "";
    }

    if (url.pathname === "/client/auto-login") {
      url.searchParams.delete("pay");
    }

    return url.toString();
  } catch {
    return link;
  }
}

File: NEXIO/backend/app/utils/pagacuotas_links.py
from urllib.parse import parse_qsl, urlencode, urlsplit, urlunsplit


def normalize_pagacuotas_portal_link(link: str | None) -> str:
    if not link:
        return ""

    try:
        parts = urlsplit(link)
        path = parts.path
        query = parts.query

        if path == "/client/payment":
            path = "/client/portal"
            query = ""
        elif path == "/client/auto-login":
            query_items = [(key, value) for key, value in parse_qsl(query, keep_blank_values=True) if key != "pay"]
            query = urlencode(query_items)

        return urlunsplit((parts.scheme, parts.netloc, path, query, parts.fragment))
    except Exception:
        return link
